lib/porcelain/drivers/goon_driver.ex

```diff
@@ -120,17 +120,17 @@ defmodule Porcelain.Driver.Goon do
 
   defp goon_options(opts) do
     ret = []
-    if opts[:in], do: ret = ["-in"|ret]
-    if opts[:out], do: ret = ["-out"|ret]
+    ret = if opts[:in], do: ["-in"|ret]
+    ret = if opts[:out], do: ["-out"|ret]
+    ret =
     case opts[:err] do
-      :out -> ret = ["-err", "out"|ret]
-      nil -> nil
```

Comment:
The new code changes the semantics. Where previously it was

> update the value of `ret` IF a specific option is mentioned

now it is

> set `ret` to a new value or `nil`

because `if` evaluates to `nil` if the condition is false and there is no `else` branch.